discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -0,0 +1,220 @@
+import ipaddress
+import os
+import shutil
+import json
+from munch import Munch
+from distutils.dir_util import copy_tree
+import logging
+from test_infra import utils
+from test_infra import virsh_cleanup
+from test_infra import consts
+from test_infra.controllers.node_controllers.libvirt_controller import LibvirtController
+
+
+class TerraformController(LibvirtController):
+
+    def __init__(self, **kwargs):
+        super(TerraformController, self).__init__(**kwargs)
+        self.cluster_name = kwargs.get('CLUSTER_NAME', f'{consts.CLUSTER_PREFIX}')
+        self.params = self.terraform_params(**kwargs)
+        self.tf_folder = self.create_tf_folder()
+        self.image_path = kwargs["ISO_DOWNLOAD_PATH"]
+
+    def __del__(self):
+        self.destroy_all_nodes()
+
+    def create_tf_folder(self):
```

Comment:
Maybe make it a private function? No one should call it except for the ctor `_create_tf_folder`

Reply:
sure